File: 1603.设计停车系统/solution.cpp
class ParkingSystem {
    int cnt[3];
public:
    ParkingSystem(int big, int medium, int small) {
        cnt[0]=big;
        cnt[1]=medium;
        cnt[2]=small;
    }
    
    bool addCar(int carType) {
        if (cnt[carType-1]==0)
            return false;
        --cnt[carType-1];
        return true;
    }
};

/**
 * Your ParkingSystem object will be instantiated and called as such:
 * ParkingSystem* obj = new ParkingSystem(big, medium, small);
 * bool param_1 = obj->addCar(carType);
 */

File: 1603.设计停车系统/solution.go
type ParkingSystem struct {
	cnt [4]int
}

func Constructor(big int, medium int, small int) ParkingSystem {
	var ans ParkingSystem
	ans.cnt[1] = big
	ans.cnt[2] = medium
	ans.cnt[3] = small
    return ans
}

func (this *ParkingSystem) AddCar(carType int) bool {
	if this.cnt[carType] > 0 {
		this.cnt[carType]--
		return true
	}
	return false
}

/**
 * Your ParkingSystem object will be instantiated and called as such:
 * obj := Constructor(big, medium, small);
 * param_1 := obj.AddCar(carType);
 */
